pkg/serverless/protocol.go

```diff
@@ -244,6 +260,14 @@ func (l *LogsCollection) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			// However, if logs are not enabled, we do not send them to the intake.
 			if sendLogsToIntake {
 				logMessage := logConfig.NewChannelMessageFromLambda([]byte(message.StringRecord), message.Time, arn, lastRequestID, functionName)
+
+				// Do not publish logs to channel if logs collection has been suspended
+				if l.daemon.logsCollectionSuspended {
```

Comment:
What happens if the logs channel is shutdown after logsCollectionSuspended is checked. I think that's another race condition. Would it be better to shutdown the Daemon HTTP API first, then shutdown the logs channel?

Reply:
Yes, I see your point here. I tried to minimize that by doing the check of `logsCollectionSuspended` at the last possible moment before we attempt to send the log message to the agent (rather than doing the check at the beginning of the API endpoint, when it would more likely be stale by the end). But I see that the race condition is not completely eliminated.

When you say shut down the Daemon HTTP API first, to clarify, I don't think we have a way to shut it down completely right now, are you proposing adding that? And then making sure that shutdown is completed before we begin the shutdown of the logs agent?

Reply:
Looks like the `http.Server` already has a `Shutdown()` method -- let me see what I can do!